gtk/package-gtk-bin.py

```diff
@@ -0,0 +1,117 @@
+
+    zstd_bin = shutil.which("zstd")
+    if args.zstd and zstd_bin is None:
+        exit("--zstd specified, but zstd command not found")
+
+    print("Packaging GTK installation for pipeline usage. This may take a while...")
+    stage_tmpfile = tempfile.TemporaryDirectory()
+    output_dir = Path(stage_tmpfile.name)
+    stage_dir = output_dir / "gtk" / "inst"
+    stage_dir.mkdir(parents=True)
+
+    def copy2_custom(src, dst):
+        """copy2 that does not follow symlinks"""
+        return shutil.copy2(src, dst, follow_symlinks=False)
+
+    def copytree_custom(*args, **kwargs):
+        return shutil.copytree(*args, **kwargs, copy_function=copy2_custom,
+                               symlinks=True)
+
+    def copy_to_stage(src_relpath: Path):
+        srcp = args.source / src_relpath
+        dstp = stage_dir / src_relpath
+        assert srcp != dstp
+        dstp.parent.mkdir(parents=True, exist_ok=True)
+        copy2_custom(srcp, dstp)
+
+    def copytree_to_stage(src_relpath: Path):
+        srcp = args.source / src_relpath
+        dstp = stage_dir / src_relpath
+        assert srcp != dstp
+        dstp.parent.mkdir(parents=True, exist_ok=True)
+        copytree_custom(srcp, dstp)
+
+    # Copy CMake
+    copy_to_stage(Path("bin") / "cmake")
+    cmake_data = list(args.source.glob("share/cmake-*"))
+    assert cmake_data, "share/cmake-* dir not found"
+    assert len(cmake_data) == 1, "more than one share/cmake-* dir unexpectedly found"
+    copytree_custom(cmake_data[0], stage_dir / "share" / cmake_data[0].name)
+
+    # Copy CTest
+    copy_to_stage(Path("bin") / "ctest")
+
+    # Copy gettext
+    gettext_bins = [
+        Path("bin") / "gettext",
+        Path("bin") / "gettext.sh",
+        Path("bin") / "xgettext",
+        Path("bin") / "msgmerge",
+        Path("bin") / "msgfmt",
+        Path("bin") / "msgcat",
+    ]
+    for p in gettext_bins:
+        copy_to_stage(p)
+    for p in args.source.glob("share/gettext*"):
+        copytree_custom(p, stage_dir / "share" / p.name)
+
+    # Copy libraries
+    copytree_to_stage("lib")
+    copytree_to_stage("include")
+    # Remove python3 as it takes a lot of space
+    for p in (stage_dir / "lib").glob("python3.*"):
+        shutil.rmtree(p)
+
+    # Copy files required by bundler
+    for p in [Path("glib-2.0") / "schemas", "locale", "themes", "icons", "gtksourceview-4"]:
+        copytree_to_stage(Path("share") / p)
+    copy_to_stage(Path("bin") / "gdk-pixbuf-query-loaders")
+    copy_to_stage(Path("bin") / "gtk-query-immodules-3.0")
+
+    subprocess.run(["find", stage_dir])
+
+    print("Generating archive")
+    if args.output.exists():
+        args.output.unlink()
+
+    if args.zstd:
+        subprocess.run([
+            "tar", "--use-compress-program", zstd_bin, "-cf", args.output,
+            "-C", output_dir, "gtk"
+        ], check=True)
+        pass
+    else:
+        subprocess.run([
+            "tar", "-cfz", args.output, "-C", output_dir, "gtk"
```

Comment:
The `f` must be at the end.
```suggestion
            "tar", "-czf", args.output, "-C", output_dir, "gtk"
```

Reply:
For uploading on Github the compressed files must be split into chunks of at most 50 MB. This was done automatically before. Do you expect the user to do it manually?

Example:
```term
split -b 50m gtk-bin.zst gtk-bin.zst.
```

Reply:
I didn't understand the reason why the split was used, thanks for clarifying. I will add a `--split` flag that can be specified to perform the splitting automatically.

Reply:
Done